cmip6_downscaling/methods/deepsd.py

```diff
@@ -0,0 +1,136 @@
+import fsspec
+import numpy as np
+import xarray as xr
+import xesmf as xe
+
+from cmip6_downscaling.workflows.paths import make_coarse_obs_path
+
+intermediate_cache_path = 'az://flow-outputs/intermediate'
+
+EPSILON = 1e-6  # small value to add to the denominator when normalizing to avoid division by 0
+INPUT_SIZE = 51  # number of pixels in a patch example used for training deepsd model (in both lat/lon (or x/y) directions)
+PATCH_STRIDE = 20  # number of pixels to skip when generating patches for deepsd training
+starting_resolutions = {
+    'GISS-E2-1-G': 2.0,
+    'BCC-CSM2-MR': 1.0,
+    'AWI-CM-1-1-MR': 1.0,
+    'BCC-ESM1': 2.0,
+    'SAM0-UNICON': 1.0,
+    'CanESM5': 2.0,
+    'MRI-ESM2-0': 1.0,
+    'MPI-ESM-1-2-HAM': 2.0,
+    'MPI-ESM1-2-HR': 1.0,
+    'MPI-ESM1-2-LR': 2.0,
+    'NESM3': 2.0,
+    'NorESM2-LM': 2.0,
+    'FGOALS-g3': 2.0,
+    'MIROC6': 1.0,
+    'ACCESS-CM2': 1.0,
+    'NorESM2-MM': 1.0,
+    'ACCESS-ESM1-5': 1.0,
+    'AWI-ESM-1-1-LR': 2.0,
+    'TaiESM1': 1.0,
+    'NorCPM1': 2.0,
+    'CMCC-ESM2': 1.0,
+}
+stacked_model_path = 'az://cmip6downscaling/training/deepsd/deepsd_models/{var}_{starting_resolution}d_to_0_25d/frozen_graph.pb'
+output_node_name = '{var}_0_25/prediction:0'
+
+
+def res_to_str(r):
+    return str(np.round(r, 2)).replace('.', '_')
+
+
+def bilinear_interpolate(output_degree, ds):
+    target_grid_ds = xe.util.grid_global(output_degree, output_degree, cf=True)
+    regridder = xe.Regridder(ds, target_grid_ds, "bilinear", extrap_method="nearest_s2d")
+    ds_regridded = regridder(ds)
+
+    return ds_regridded
+
+
+def conservative_interpolate(output_degree, ds):
```

Comment:
since this one isn't used should it be removed? is there any world in which we might want to switch to it?

Reply:
oh i see it's used in the data_prep notebook. nevermind!